mqtt/mqtt-edgehub/src/settings.rs

```diff
@@ -75,8 +76,83 @@ impl Source for BrokerEnvironment {
+#[derive(Debug, Clone)]
+pub struct BridgeEnvironment;
+
+impl Source for BridgeEnvironment {
+    fn clone_into_box(&self) -> Box<dyn Source + Send + Sync> {
+        Box::new((*self).clone())
+    }
+
+    // BridgeEnvironment allows only the env variables explicitly
+    // defined in the method below.
+    //
+    // We use intermediate instance of `Config` to enumerate all env vars
+    // and then manually map them to our internal config structure.
+    // This is done for two reasons:
+    // - our bridge config structure does not match legacy EdgeHub env vars,
+    // - `Config` does a bunch of useful things - takes care of
+    //   evn vars casing, prefixing, separators...
+    //
+    // NOTE: if adding new env vars - don't forget to use lowercase
+    // and update `check_env_var_can_override_bridge_settings` test.
+    fn collect(&self) -> Result<HashMap<String, Value>, ConfigError> {
+        let mut host_env = Config::new();
+        // regular env vars
+        host_env.merge(Environment::new())?;
+        // broker specific vars
+        host_env.merge(Environment::with_prefix("MqttBridge_").separator(":"))?;
+        host_env.merge(Environment::with_prefix("MqttBridge_").separator("__"))?;
+
+        let mut result: HashMap<String, config::Value> = HashMap::new();
+
+        // edgehub module upstream settings
+        if let Ok(val) = host_env.get::<Value>("iotedge_iothubhostname") {
+            result.insert("bridge.iothub_hostname".into(), val);
+        }
+        if let Ok(val) = host_env.get::<Value>("iotedge_gatewayhostname") {
+            result.insert("bridge.gateway_hostname".into(), val);
+        }
+        if let Ok(val) = host_env.get::<Value>("iotedge_deviceid") {
+            result.insert("bridge.device_id".into(), val);
+        }
+        if let Ok(val) = host_env.get::<Value>("iotedge_moduleid") {
+            result.insert("bridge.module_id".into(), val);
+        }
+        if let Ok(val) = host_env.get::<Value>("iotedge_modulegenerationid") {
+            result.insert("bridge.generation_id".into(), val);
+        }
+        if let Ok(val) = host_env.get::<Value>("iotedge_workloaduri") {
+            result.insert("bridge.workload_uri".into(), val);
+        }
+
+        // storage rign buffer
+        if let Ok(val) = host_env.get::<Value>("storagetype") {
+            result.insert("bridge.storage.type".into(), val);
+        }
+        if let Ok(val) = host_env.get::<Value>("storagemaxfilesize") {
+            result.insert("bridge.storage.max_file_size".into(), val);
+        }
+        if let Ok(val) = host_env.get::<Value>("storageflushoptions") {
+            result.insert("bridge.storage.flush_options".into(), val);
+        }
+        if let Ok(val) = host_env.get::<Value>("storagefolder") {
+            result.insert("bridge.storage.directory".into(), val);
+        }
+
+        // storage in memory
+        if let Ok(val) = host_env.get::<Value>("storagesize") {
```

Comment:
```suggestion
        if let Ok(val) = host_env.get::<Value>("storagememorysize") {
```
As a user, I'm confused with MaxStorageSize and MaxStorageFileSize. Why not use only one value to rule them all? :)
`MaxStorageMemorySize` and `MaxStorageFileSize` look cleaner

Reply:
At least when I made the changes for settings on memory and ringbuffer, I wanted it to be clear that the storage values have slightly different side effects, i.e. memory vs disk. I think combining into one value might look nicer and should be hard to destroy in Mordor 😸 Let's just make sure we call out somewhere in the documentation that depending on storage type these can use different resources?

Reply:
@dmolokanov, what you say makes sense, but one thing to keep in mind, is that those two variables are actually different units. Storage file size is in bytes (or Humansize), where memory size is in number of publications (not bytes). So I believe combining them will be confusing.